core/vm/gas_table.go

```diff
@@ -350,14 +351,37 @@ func pureMemoryGascost(evm *EVM, contract *Contract, stack *Stack, mem *Memory,
 }
 
 var (
-	gasReturn  = pureMemoryGascost
-	gasRevert  = pureMemoryGascost
-	gasMLoad   = pureMemoryGascost
-	gasMStore8 = pureMemoryGascost
-	gasMStore  = pureMemoryGascost
-	gasCreate  = pureMemoryGascost
+	gasReturn         = pureMemoryGascost
+	gasRevert         = pureMemoryGascost
+	gasMLoad          = pureMemoryGascost
+	gasMStore8        = pureMemoryGascost
+	gasMStore         = pureMemoryGascost
+	statefulGasCreate = pureMemoryGascost
 )
 
+func gasCreate(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
+	var gasUsed uint64
+	if gasUsed, err := statefulGasCreate(evm, contract, stack, mem, memorySize); err != nil {
+		return gasUsed, err
+	}
+
+	if evm.chainConfig.IsCancun(evm.Context.BlockNumber) {
+		var overflow bool
+		statelessGas := evm.Accesses.TouchAndChargeContractCreateInit(contract.Address().Bytes()[:])
```

Comment:
This charging is repeated `core/vm/instructions.go` (also for `CREATE2`.  Should be removed from `core/vm/gas_table.go`.  I'll do it tomorrow.